packages/fxa-admin-panel/src/components/PageRelyingParties/index.tsx

```diff
@@ -34,6 +37,93 @@ export const GET_RELYING_PARTIES = gql`
   }
 `;
 
+export const UPDATE_NOTE = gql`
+  mutation updateNotes($id: String!, $notes: String!) {
+    updateNotes(id: $id, notes: $notes)
+  }
+`;
+
+const Notes = ({ id, notes }: { id: string; notes: string }) => {
+  const [saveBtn, setSaveBtn] = useState<string>('Save');
+  const [saveVis, setSaveVis] = useState<boolean>(false);
+  const [error, setError] = useState<string>('');
+  const [newNotes, setNewNotes] = useState<string>(notes);
+
+  const isSaveDisabled = () => {
+    return !saveVis || error !== '' || saveBtn !== 'Save';
+  };
+
+  const [updateNote] = useMutation(UPDATE_NOTE, {
+    onCompleted: (data) => {
+      setSaveBtn('Success!');
+      setTimeout(() => {
+        setSaveVis(false);
+      }, 1000);
+    },
+
+    onError: (err) => {
+      let error = 'Error: Unexpected Error.';
+      if (/ER_DATA_TOO_LONG/.test(err.message)) {
+        error = 'Error: Changes not saved. Notes too long!';
+      }
+      setSaveVis(false);
+      setError(error);
+    },
+  });
+
+  const handleNotesChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    if (event.target.value === notes) {
+      return;
+    }
+    setSaveBtn('Save');
+    setSaveVis(true);
+    setError('');
+    setNewNotes(event.target.value);
+  };
+
+  const handleSaveNotes = () => {
+    setSaveBtn('Updating...');
+    updateNote({
+      variables: {
+        id,
+        notes: newNotes,
+      },
+    });
+  };
+
+  return (
+    <>
+      <textarea
+        data-testid={`notes-${id}`}
+        className="w-full mt-4 mb-2 border border-grey-100"
+        onChange={handleNotesChange}
+        defaultValue={notes}
+      ></textarea>
+      <Guard features={[AdminPanelFeature.RelyingPartiesEditNotes]}>
+        <br />
+        <button
+          data-testid={`notes-save-btn-${id}`}
+          className="bg-grey-100 px-3 rounded-r flex flex-row items-center transition duration-800"
+          style={{
+            opacity: saveVis ? 1 : 0,
+          }}
```

Comment:
Ah, I like the className approach better. Not sure why I didn't think of that. Gotta get more familiar with my tailwind classes. 

I sort of like the effect of hiding the button, cause when they are all there it just seemed a bit redundant... I really wasn't sure what had better UX though. If you think using cta-secondary is better, I'd be happy to change it.

Reply:
Yeah, I agree the extra space needs to be there for the fade out/fade in effect to avoid it jumping. If you want to send me a screenshot with `cta-neutral` (not `cta-secondary` lol oops) I can tell you my thoughts but I'm fine with whichever you like better. 😄

Reply:
I've updated this so the save button is always shown and has the style as other admin panel buttons. So I think it's good to go now.